interface/backend/images/models.py

```diff
@@ -1,18 +1,147 @@
-import glob
+import base64
+import logging
+import os
 
 import dicom
+import numpy as np
+
+from django.conf import settings
 from django.db import models
+from django.core.exceptions import PermissionDenied
 from django.utils._os import safe_join
 
 
+class ImageFile(models.Model):
+    """ Model for an individual file for an image--that is, a single DICOM slice
+        as it is represented on disk.
+    """
+    # Explicitly map all the dicom properties that we need to parse and their
+    # model equivalents; create a lambda that returns a dict for any type conversions
+    # or one-to-many relationships
+    DICOM_PROPERTIES = {
+        'SliceLocation': lambda x: {'slice_location': float(x)},
+        'SliceThickness': lambda x: {'slice_thickness': float(x)},
+        'RescaleSlope': lambda x: {'rescale_slope': int(x)},
+        'Rows': 'rows',
+        'Columns': 'columns',
+        'PixelSpacing': lambda x: {'pixel_spacing_col': float(x[0]), 'pixel_spacing_row': float(x[1])},
+    }
+
+    series = models.ForeignKey('ImageSeries', related_name='images', on_delete=models.CASCADE)
+    slice_location = models.FloatField(null=True)
+    slice_thickness = models.FloatField(null=True)
+    rescale_slope = models.IntegerField(null=True)
+    rows = models.IntegerField(null=True)
+    columns = models.IntegerField(null=True)
+    pixel_spacing_col = models.FloatField(null=True)
+    pixel_spacing_row = models.FloatField(null=True)
+    path = models.FilePathField(path=settings.DATASOURCE_DIR,
+                                recursive=True,
+                                allow_files=True,
+                                allow_folders=False,
+                                max_length=512)
+
+    class Meta:
+        ordering = ['slice_location']
+
+    def save(self, *args, **kwargs):
+        self._populate_dicom_properties()
+        super().save(*args, **kwargs)
+
+    def _populate_dicom_properties(self):
+        """ Parse DICOM properties from the file and store the relevant ones
+            on the object.
+        """
+        logger = logging.getLogger(__name__)
+        for k, v in self.load_dicom_data_from_disk(self.path)['metadata'].items():
+            # check that the field is a property on the object
+            if not hasattr(self.__class__, k):
+                logger.warning(f"Property '{k}' not a field on ImageFile, discarding value '{v}'.")
+
+            else:
+                self.__setattr__(k, v)
+
+    @classmethod
+    def load_dicom_data_from_disk(cls, filepath, parse_metadata=True, encode_image_data=False):
```

Comment:
Maybe a just a matter of style, but based on class methods i've seen how about simplifying to just `from_disk`? This assumes `ImageFile` is only for DICOM files of course, otherwise something like `dicom_from_disk` would work.